sklearn/random_projection.py

```diff
@@ -217,7 +217,7 @@ def _sparse_random_matrix(n_components, n_features, density='auto',
     n_features : int,
         Dimensionality of the original source space.
 
-    density : float in range ]0, 1] or 'auto', optional (default='auto')
+    density : float in range (0, 1] or 'auto', default='auto'
```

Comment:
```suggestion
    density : float or 'auto', default='auto'
```

Could you add the information regarding the range in the description below the type.